### ISSUE TITLE
WLED Compatibility

### ISSUE THREAD

**Issue by guiggess** [Blackymas/NSPanel_HA_Blueprint#1307]
Hello there,
I am no developper and can only thank Blackymas and the community for all the work done to make this blueprint what it is today!
I'm a user of WLED, and would love to have some more feature on the panel for such an entity (ie. Presets / Playlist selection, Nighlight, intensity, etc...)
Thanks

**Comment by luckiesdavid** [comment]
Hi, i also have WLED and  for my use most functions are available, what functions do you mean?

**Comment by wwebers** [comment]
I am also using WLED for several of my lamps. Would also like to be able to choose more than just color.
I wonder if the Nextion displays allow such things like dropdown lists though.

**Comment by edwardtfn** [comment]
The problem is exactly that this model of Nextion display used in the NSPanel doesn't have a native dropdown component, so we have to find some time to create this component.
I wanna do this, as it will be useful not only for this, but many other cases, however I'm quite busy right now and cannot promise any date. 😞

**Comment by wwebers** [comment]
@edwardtfn Is this definition of a dropdown list done inside the Nextion editor? Never used this editor, but am curious.

**Comment by edwardtfn** [comment]
Yes, we have to create all the visual things to show the values, expand in a list, buttons, etc. and all the code related to that.
New versions of Nextion already have this component, so I might be able to copy a bit the UI and logic, but it will require some time anyways.

**Comment by guiggess** [comment]
> The problem is exactly that this model of Nextion display used in the NSPanel doesn't have a native dropdown component, so we have to find some time to create this component. I wanna do this, as it will be useful not only for this, but many other cases, however I'm quite busy right now and cannot promise any date. 😞

No rush mate, it’s a nice feature to me

**Comment by PoopingNow** [comment]
I am very thankful for all of your efforts. I too would like to see a drop down component. No rush, I can wait. I know it's not an easy task.

**Comment by dimis7** [comment]
+1 for a WLED integration (drop down menu for presets / playlist / effects maybe?)

nspanel-lovelace-ui can show drop down menus, so it should be possible to take part of the code / ideas from there?

**Comment by wwebers** [comment]
Aside that they work with a different architecture based on a backend (like HA) and just display things, I cannot find any HMI definition of drop down lists there either.

Maybe you could provide a link to the code you mention?

**Comment by dimis7** [comment]
Hi!

I flash again the nspanel-lovelace-ui firmware because i remember it was working...

The truth is that on the NSpanel itself there is no drop-down menu.
But when you chose a "select" type entity, a new page is displayed with buttons that you can use to set desired value.

For example: 
The preset entity from HA is like this:

![Image](https://github.com/user-attachments/assets/b8a4663e-9df0-4e13-8824-79ae9b3f2e74)

On the NSPanel is like this:

![Image](https://github.com/user-attachments/assets/ac0b0015-373f-4d43-b377-83898520bd7d)

When you click on "Presets" button the following page is displayed:

![Image](https://github.com/user-attachments/assets/0631035f-30f0-41e6-8dc5-2052f277b7c1)


Its not perfect, but it's working. Just an idea of course!

**Comment by wwebers** [comment]
Ah, I see. But that's not really a dropdown but more an own page. Would be interesting how they realize scrolling in case you have 100 items in the list, like WLED has for its effects for example. Not sure if that's really that user friendly, maybe for a very limited amount of items like in your example.

Its a shame that this panel is so limited (and slow) when it comes to its UI capabilties compared to their Android based panel (which is overkill in my eyes).
